odoo/addons/base/models/ir_ui_view.py

```diff
@@ -532,10 +534,8 @@ def write(self, vals):
         # and therefore when writing `active` and `arch` at the same time, `_check_xml` is called twice,
         # and the first time it tries to validate the view without the modification to the arch,
         # which is problematic if the user corrects the view at the same time he re-enables it.
-        if vals.get('active'):
```

Comment:
It is the same behavior than before, but maybe we should add `or 'inherit_id' in vals` because IMO, it should also trigger `_check_xml`.